### ISSUE TITLE
Processing algorithms fail to give a proper name to output layers

### ISSUE THREAD

**Issue by SrNetoChan** [qgis/qgis3_UIX_discussion#20]
When finished processing, some algorithms fail to give a proper name to the output files that are loaded in the map canvas, even if the user choose to save the output to a file. But other seem to work well.

It would be nice to hunt down this kind of algorithms as introduce this missing functionality.

Extra: for output saved in memory, it would be nice to add someking of suffix, to avoid repeat layer names. Ex: output_1, output_2.

**Comment by SrNetoChan** [comment]
@alexbruy, am I missing something here?

**Comment by pcav** [comment]
Agreed with the last point (adding a serial to layer name), I think this is useful for all types of layers.

**Comment by anitagraser** [comment]
If the users did chose a file name, I think the file name should always be used in the layer list - irrespective of the "use filename as layer name" option in the settings. 

We could even drop the "use filename as layer name" option. Behavior would be:

If temporary file path is used --> use algorithm default output layer name
If user-defined path is used --> use filename

**Comment by SrNetoChan** [comment]
@anitagraser I don't mind having that option, as long as is set as default.

Not sure why someone would want to not use the filename as layer name, but there might be som user case.

**Comment by anitagraser** [comment]
> Not sure why someone would want to not use the filename as layer name

Well, if you take the time to manually pick and write a file name, why wouldn't you want it as the layer name?
